### PR TITLE
Add a form to hide personal data on company page

### LINKED ISSUE #676: [Support] non-diffusibles 2/2 : créer un formulaire de demande de protection des données d'entreprise
## Process actuel : 

Les non diffusibles peuvent directement changer leur statut de diffusion grace au site de l'Insee

Les autres entreprises peuvent nous demander a faire disparaitre leur page dirigeant et être desindéxée

Le page FAQ explique comment faire : https://annuaire-entreprises.data.gouv.fr/faq/supprimer-donnees-personnelles-entreprise

Le dirigeant nous écrit et on ajoute son siren a la liste sur grist

### Problèmes 
- pas de vérification de l'identité
- pas de cadrage clair (un EI peut nous écrire, ou bien un dirigeant peut oublier d'indiquer son siren etc.)
- pollution du support

### Pour l'utilisateur

Remplacer le mail par un formulaire dédié page : 
- FranceConnecté (vu avec le pole juridique)
  - si FC échoue -> indiquer le mail 
- Champ de saisie de siren
- possibilité de mettre plusieurs siren
- vérification
  - si EI -> message de redirection vers l'Insee
  - si déja dans protected siren -> message d'erreur

A la soumission -> créer une table temporaire de gestion dans GRIST
avec le siren, le nom, prenom de l'utilisateur qui fait la demande et un lien préformaté vers la page dirigeant

Ou alors : si l'etat civil match -> donner directement gain de cause, sinon table temporaire ? A discuter

### Pour le support 
 
Ouvrir la page dirigeant, vérifier que ca match et si ca match -> ajouter le siren a la table officielle dans grist et supprimer la ligne

A voir si on peut automatiser l'action avec un script dans grist

### references 

https://github.com/etalab/annuaire-entreprises-site/pull/73

### PR DESCRIPTION
**Pull request by johangirod**
Fix #676 

## Todo 
- [x] Change FAQ
- [x] Add env variables to prod & scalingo
- [x] Recette FranceConnect
- [x] Change URL
- [x] Integrate FranceConnect
- [x] Create a table in Grist
- [x] Manualluy test all cases
- [x] Add a logout functionality

### CODE REVIEW

**Review by XavierJp — COMMENTED** [review]
Very nice.

Things I really like : 
- POST approach is concise and effective
- a11y !
- model is very clear
- love the ability to check on past requests
- maybe past requested siren could be store in local storage ?

Things I have questions : 
- `pages/faq/demande-non-diffusible/index.tsx`do a lot. it is a page, a form and handle rendering. I think moving html into dedicated components can greatly improve readibility
- we miss the ability to create multiple requests. Some people have 5+ companies. We need to ask Anais if this is ok
- I wonder if siren Input would not be better placed before FC (no need be connected to display redirect EI message or to search for siren)
- I feel like react would be a plus for the siren input. A simple usage of the API could help users to find their siren

Would happily discuss it this afternoon !

File: clients/auth/france-connect/strategy.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  if (!req.session.franceConnect?.tokenId) {
+    throw new InternalError({ message: 'No token id' });
+  }
+  const client = await getClient();
+
+  const state = `state${randomBytes(32).toString('hex')}`;
+  const id_token_hint = req.session.franceConnect.tokenId;
+  delete req.session.franceConnect;
+  await req.session.save();
+  console.log({
```
Comment (johangirod) [inline clients/auth/france-connect/strategy.ts:112]: Enlever le console.log

**Review by XavierJp — APPROVED** [review]
Overrinding FAQ path with `pages/faq/supprimer-donnees-personnelles-entreprise/index.tsx` seem to break the current FAQ logic and therefore it does not appear anymore in FAQ : https://annuaire-entreprises-site-pr838.osc-fr1.scalingo.io/faq/parcours

-> maybe we can create a dedicated page ? And have FAQ make a link to this page ?

Interface is simpler without the stepper 👍 

Currently the FC button + input + greyed validation is a bit overwhelming 

Sorry if my feedback were confusing, I thought we nee to get rid of the component stepper, but I believe we still need two steps : 

1. authentification 
2. siren input

Note that it could be the other way around. It would work

Also "→ Rechercher le SIREN de mon entreprise" could use a target="_blank" and open a new tab otherwise you might loose your page wich can be frustrating

**Comment by XavierJp** [comment]
I think we should also add a link to this page in our footer

**Comment by XavierJp** [comment]
I think we can simplify the screens :
image

image

**Comment by XavierJp** [comment]
Despite my feedbacks I think we can already merge this PR, this is only fine tuning

**Comment by johangirod** [comment]
@XavierJp j'ai pris en compte tes retours : https://annuaire-entreprises-site-pr838.osc-fr1.scalingo.io/faq/supprimer-donnees-personnelles-entreprise

En revanche, je n'ai pas créé une nouvelle page, je n'en vois pas la nécessité (j'ai corrigé le bug qui la faisait disparaitre du parcours).

La demande de recettage est envoyée, il n'y a plus qu'à patienter.

**Comment by XavierJp** [comment]
I took FC team feedbacks into account : 
- added a redirection on logout
- also added links in privacy page
- harmonized a bit components to use existing ones when possible
- resolved conflicts

**Pull request merged**